dbms/src/Debug/astToExecutor.cpp

```diff
@@ -1648,30 +1657,64 @@ ExecutorPtr compileProject(ExecutorPtr input, size_t & executor_index, ASTPtr se
     return project;
 }
 
-ExecutorPtr compileJoin(size_t & executor_index, ExecutorPtr left, ExecutorPtr right, ASTPtr params)
+ExecutorPtr compileJoin(size_t & executor_index, ExecutorPtr left, ExecutorPtr right, tipb::JoinType tp, ASTPtr using_expr_list)
 {
     DAGSchema output_schema;
-    const auto & join_params = (static_cast<const ASTTableJoin &>(*params));
+
     for (auto & field : left->output_schema)
```

Comment:
maybe split the schema building process into “buildLeftSchema” and “buildRightSchema”？